Add github to check and automatically update the Zarf Injector component upon renovate execution (or other PR)

mcamick opened the pull request:
Was messing with renovate and other tooling to see if I could get the Zarf injector to update automatically. Since the file change is in the zarf-config.toml file upstream and not in the release itself or tagging, there's nothing I think renovate can do. So, I think a Github action to pull the file and check is still a decent option. I forked the repo and tested it by changing the values and watching the action change it back on the push to main.

ntwkninja commented on .github/workflows/update-zarf-injector.yaml:
```diff
@@ -0,0 +1,59 @@
+
+      - name: Fetch zarf-config.toml from zarf-dev
+        run: |
+          git clone https://github.com/zarf-dev/zarf.git
```
We should `git checkout ${zarf_version}` before running `cp` on the file 

zarf_version=[this](https://github.com/defenseunicorns/delivery-zarf-init/blob/main/zarf-config.yaml#L9) field)?

mcamick replied:
I think it's resolved now?

ntwkninja commented on .github/workflows/update-zarf-injector.yaml:
```diff
@@ -0,0 +1,88 @@
+name: Zarf Injector Update
+
+on:
+  pull_request:
```
Because this repo is public, I think that adding an on pull request workflow will allow any fork to run arbitrary code (unless we have stringent repo/branch settings).

We may want the update injector workflow to work similar to the publish workflow, `on push to main` 
- check if it needs updating, then update else exit 

cc: @bburky

mcamick replied:
I removed the additional logic, the original intent was to have it work alongside renovate prior to publishing.

mcamick commented:
Okay, I think I fixed the action logic. I moved the original logic for release into the injector update, so it only triggers on push to main where the zarf.yaml or zarf-config.yaml changes and then the release action triggers on completion of that job. This should address the security concerns and finalize the feature with intended functionality.
